tensorboard/pip_package/pip_smoke_test.sh

```diff
@@ -0,0 +1,183 @@
+echo
+
+# If virtualenv is not installed, install it.
+if [[ -z $(which virtualenv) ]]; then
+  echo
+  echo "Installing virtualenv"
+  echo
+  pip install virtualenv
+fi
+
+if [[ ${PY_VERSION} == 2 ]]; then
+  pip install wheel
+else
+  pip3 install wheel
+fi
+
+PIP_TMP_DIR=$(mktemp -d --suffix _tensorboard)
+
+echo
+echo "Building tensorboard pip package in directory: ${PIP_TMP_DIR}"
+echo
+
+bazel build -c opt tensorboard/pip_package:build_pip_package
+bazel-bin/tensorboard/pip_package/build_pip_package "${PIP_TMP_DIR}"
+
+if [[ ${PY_VERSION} == 2 ]]; then
+  WHL_PATH=$(ls ${PIP_TMP_DIR}/tensorflow_tensorboard*-py2-*.whl)
+elif [[ ${PY_VERSION} == 3 ]]; then
+  WHL_PATH=$(ls ${PIP_TMP_DIR}/tensorflow_tensorboard*-py3-*.whl)
+fi
+
+if [[ -z ${WHL_PATH} ]]; then
+  die "ERROR: Cannot find the built pip package for python version ${PY_VERSION}"
+else
+  echo
+  echo "Found the built pip package for python version ${PY_VERSION} at:"
+  echo "  ${WHL_PATH}"
+  echo
+fi
+
+# Create virtualenv directory, cleanly (i.e., no --system-site-packages).
+VENV_TMP_DIR=$(mktemp -d --suffix _tensorboard_venv)
+
+echo
+echo "Creating virtualenv directory at: ${VENV_TMP_DIR}"
+echo
+
+if [[ ${PY_VERSION} == 2 ]]; then
+  PYTHON_PATH="/usr/bin/python"
+  TF_NIGHTLY_URL='https://ci.tensorflow.org/view/Nightly/job/nightly-matrix-cpu/TF_BUILD_IS_OPT=OPT,TF_BUILD_IS_PIP=PIP,TF_BUILD_PYTHON_VERSION=PYTHON2,label=cpu-slave/lastSuccessfulBuild/artifact/pip_test/whl/tensorflow-1.head-cp27-none-linux_x86_64.whl'
+elif [[ ${PY_VERSION} == 3 ]]; then
+  PYTHON_PATH="/usr/bin/python3"
+  TF_NIGHTLY_URL='https://ci.tensorflow.org/view/Nightly/job/nightly-matrix-cpu/TF_BUILD_IS_OPT=OPT,TF_BUILD_IS_PIP=PIP,TF_BUILD_PYTHON_VERSION=PYTHON3,label=cpu-slave/lastSuccessfulBuild/artifact/pip_test/whl/tensorflow-1.head-cp34-cp34m-linux_x86_64.whl'
+fi
+
+virtualenv -p ${PYTHON_PATH} "${VENV_TMP_DIR}"
+source ${VENV_TMP_DIR}/bin/activate
+
+# Install the dependency, tensorflow, first.
+echo
+echo "Installing nightly tensorflow pip package."
+echo
+
+pip install ${TF_NIGHTLY_URL}
+
+echo
+echo "Installing the just-built tensorboard pi package."
+echo
+
+pip install ${WHL_PATH}
+
+# Check tensorboard binary path.
+TB_BIN_PATH=$(which tensorboard)
+if [[ -z ${TB_BIN_PATH} ]]; then
+  die "ERROR: Cannot find tensorboard binary path after installing tensorboard "\
+"pip package."
+fi
+
+tensorboard --port=${TEST_PORT} --logdir=/tmp/foo &
+TB_PID=$!
+
+TEST_URL="http://localhost:${TEST_PORT}"
+echo
+echo "tensorboard binary should be running at pid ${TB_PID}"
+echo "Sending test HTTP requests at URL: ${TEST_URL} (${NUM_RETRIES} retries)"
+echo
+
+set +e
+
+RETRY_COUNTER=0
+REQUEST_SUCCEEDED=0
+while [[ ${RETRY_COUNTER} -lt ${NUM_RETRIES} ]]; do
+  sleep 1
+  STATUS_CODE=$(curl -Is "${TEST_URL}" | head -1 | cut -d ' ' -f 2)
+  if [[ ${STATUS_CODE} == 200 ]]; then
+    echo
+    echo "Request succeeded (200)!"
+    echo
+    REQUEST_SUCCEEDED=1
+    break;
+  else
+    ((RETRY_COUNTER++))
```

Comment:
Google-style requires that you replace this with `: $(( RETRY_COUNTER++ ))` (although what you have happens to be fine in this case).